pkg/operator/ceph/csi/peermap/config.go

```diff
@@ -0,0 +1,337 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peermap
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/coreos/pkg/capnslog"
+	"github.com/pkg/errors"
+	cephv1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	cephclient "github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/operator/k8sutil"
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	mappingConfigName = "rook-ceph-csi-mapping-config"
+	mappingConfigkey  = "csi-mapping-config-json"
+)
+
+var logger = capnslog.NewPackageLogger("github.com/rook/rook", "peer-map")
+
+type PeerIDMapping struct {
+	ClusterIDMapping map[string]string
+	RBDPoolIDMapping []map[string]string
+}
+
+type PeerIDMappings []PeerIDMapping
+
+// addClusterIDMapping adds cluster ID map if not present already
+func (m *PeerIDMappings) addClusterIDMapping(newClusterIDMap map[string]string) {
+	if m.clusterIDMapIndex(newClusterIDMap) == -1 {
+		newDRMap := PeerIDMapping{
+			ClusterIDMapping: newClusterIDMap,
+		}
+		*m = append(*m, newDRMap)
+	}
+}
+
+//addRBDPoolIDMapping adds all the pool ID maps for a given cluster ID map
+func (m *PeerIDMappings) addRBDPoolIDMapping(clusterIDMap, newPoolIDMap map[string]string) {
+	for i := 0; i < len(*m); i++ {
+		if reflect.DeepEqual((*m)[i].ClusterIDMapping, clusterIDMap) {
+			(*m)[i].RBDPoolIDMapping = append((*m)[i].RBDPoolIDMapping, newPoolIDMap)
+		}
+	}
+}
+
+func (m *PeerIDMappings) clusterIDMapIndex(newClusterIDMap map[string]string) int {
+	for i, mapping := range *m {
+		if reflect.DeepEqual(mapping.ClusterIDMapping, newClusterIDMap) {
+			return i
+		}
+	}
+	return -1
+}
+
+func (m *PeerIDMappings) toString() (string, error) {
```

Comment:
```suggestion
func (m *PeerIDMappings) String() (string, error) {
```

I think this is more common